aiida_quantumespresso/parsers/parse_raw/base.py

```diff
@@ -133,29 +140,33 @@ def convert_qe_time_to_sec(timestr):
     return num_seconds
 
 
-def convert_qe2aiida_structure(output_dict, input_structure=None):
-    """Receives the dictionary cell parsed from quantum espresso Convert it into an AiiDA structure object."""
-    from aiida.plugins import DataFactory
-
-    StructureData = DataFactory('structure')
+def convert_qe_to_aiida_structure(output_dict, input_structure=None):
+    """Convert the dictionary parsed from the Quantum ESPRESSO output into ``StructureData``."""
 
     cell_dict = output_dict['cell']
 
-    # If I don't have any help, I will set up the cell as it is in QE
+    # Without an input structure, try to recreate the structure from the output
     if not input_structure:
 
-        s = StructureData(cell=cell_dict['lattice_vectors'])
-        for atom in cell_dict['atoms']:
-            s.append_atom(position=tuple(atom[1]), symbols=[atom[0]])
+        structure = StructureData(cell=cell_dict['lattice_vectors'])
+
+        for mass, name in zip(output_dict['species']['mass'], output_dict['species']['type']):
+
+            # Remove digits from the kind name to obtain the symbol
+            symbol = re.sub('\d+', '', name)
+            structure.append_kind(Kind(symbols=symbol, mass=mass, name=name))
+
+        for kind_name, position in output_dict['atoms']:
+            structure.append_site(Site(kind_name=kind_name, position=position))
```

Comment:
I don't think you have to mess with the masses your self. You can simply define the kind name with the `name` argument in `append_atom`. It will figure out the correct `Kind`s itself
```suggestion
        for kind_name, position in output_dict['atoms']:
            symbol = re.sub('\d+', '', kind_name)
            structure.append_atom(position=position, symbols=symbol, name=kind_name))
```